File: src/main/java/dominio/indicadores/ExpresionIndicador.java
package dominio.indicadores;

import dominio.empresas.Empresa;
import dominio.usuarios.Usuario;

import java.time.Year;

public class ExpresionIndicador implements Expresion{
	
	private String nombreIndicador;
	
	public ExpresionIndicador(String nombreIndicador){
		this.nombreIndicador=nombreIndicador;		
	}
	
	public int evaluarEn(Empresa empresa, Year anio){
		Indicador indicador = Usuario.activo().buscarIndicador(nombreIndicador);
		return indicador.evaluarEn(empresa,anio);
	}
	
}

File: src/main/java/dondeInvierto/dominio/indicadores/ExpresionIndicador.kt
package dondeInvierto.dominio.indicadores

import dondeInvierto.dominio.empresas.Empresa
import dondeInvierto.dominio.Usuario

import java.time.Year

class ExpresionIndicador(private val nombreIndicador: String) : Expresion {

    override fun evaluarEn(empresa: Empresa, anio: Year): Int {
        val indicador = Usuario.activo!!.buscarIndicador(nombreIndicador)
        return indicador.evaluarEn(empresa, anio)
    }

}
